### ISSUE TITLE
Nvidia Shield Pro HDR and 5.1 Problem

### ISSUE THREAD

**Issue by JohnJayRafka** [yuliskov/SmartTubeNext#1301]
good evening, first of all I just want to say that is a great app!!  I have a small problem with the Nvidia Shield Pro, although HDR and 5.1 no longer work.  on my fire tv cube has HDR and 5.1 works on their app.  I don't know if you already know the problem?  and wanted to let you know maybe they can somehow fix it in the next update or later.  Thank you very much for this app and of course I will donate something for this great app, keep it up.

**Comment by finnzz** [comment]
The Shield doesn't support VP9 profile2 which is used for HDR on YouTube. So no HDR on the Shield.  FireTV devices do support VP9 profile 2

**Comment by Thlerry** [comment]
How about 5.1 sound ?

It seems Youtube app on other platforms (LG TV, Apple TV) supports 5.1 now, however Youtube/Stn on Nvidia Shield Pro 2019 always plays 2 channel PCM even for video with 5.1 surround sound.

**Comment by finnzz** [comment]
5.1 audio is officially supported on YouTube TV.  The official YouTube app only officially supports stereo on all platforms.  

The official YouTube app did briefly support 5.1 audio a couple years ago, but then reversed that.  SmartTubeNext is able to access a very very few number of videos with audio tracks labeled 5.1.  This would be AAC 5.1 audio, and you would need a receiver that supports AAC 5.1 decoding. 

I don't know what variants of 5.1 mp4a the Shield supports passthrough for.  The Shield Reddit sub would be a lot more helpful there.


 official YouTube app only supports stereo no matter the platform. YouTube TV has officially added support for 5.1 audio.  

What YouTube unofficially supports is murky.  The regular YouTube app did add 5.1 audio support a couple years ago, but then they removed it.  You can find a very very limited number of videos through

**Comment by Thlerry** [comment]
If you look at the following video for instance,  it plays as Dolby 5.1 on Apple TV and LG TV with the YouTube app:

https://www.youtube.com/watch?v=0549HF0BtRo&feature=youtu.be

**Comment by finnzz** [comment]
I don't have the an LG or AppleTV to test their YouTube apps.  Those apps may indeed be accessing the mp4a 5.1 audio track, the same one you can see in SmartTubeNext.

My guess is that because AAC 5.1 audio can't be passed through optical or ARC, that both the LG and AppleTV are transcoding the mp4a 5.1 track to AC3 5.1.  I doubt there is an actual AC3 track on YouTube's server. If someone knows otherwise, please correct me.

If you want want SmartTubeNext to transcode mp4a 5.1 to AC3 5.1, maybe @yuliskov can do that?  I don't know how easy that can be done considering the hardware restraints on a lot of AndroidTV devices.

**Comment by JohnJayRafka** [comment]
my shield is connected to my denon x1700h.  The Denon is connected to my LG via eARC.  I have to see if the shield can encode it.  but isn't everything then coded even if I don't use smartube?

**Comment by finnzz** [comment]
I was searching for a list of supported codecs for the Denon x1700H, and it lists AAC, but not AAC5.1 explicitly.  So I'm not sure if it does or doesn't support that.  You should get a video with AAC 5.1 audio and try to direct play it through Plex to see if your receiver reports AAC 5.1 or PCM 5.1.

Because you have eARC support, your best option assuming the receiver can't decode multichannel AAC, is for the Shield to convert AAC 5.1 to PCM 5.1 (lossless) and pass that through to the receiver.

Since PCM is lossless, converting AAC to PCM will not cost you any sound quality.

**Comment by Thlerry** [comment]
I have a Marantz SR6015, and Shield Pro 2019 connected to it directly.
Kodi with passthrough enabled plays AAC 5.1 as PCM Multi Channel. I don't think AAC is widely supported in AVRs (probably close to none).

AAC usually gets played as PCM Multi Channel, or gets translated to DD5.1 by some.
So I guess playing 5.1 as PCM Multi Channel is probably the easiest when supported by the device.

I think you are right in case of Apple TV and LG TV the app itself transcodes to DD 5.1. Do you know if in the link I sent the audio track is Dolby or AAC ? It's weird that they both play as Dolby since the ATV would normally decode AAC to PCM Multichannel (Like Infuse does for instance).

**Comment by finnzz** [comment]
When I open your link in SmartTubeNext I see the mp4a 5.1 track which is AAC 5.1. I agree it's pretty hard to find any receivers or soundbars that support AAC 5.1.  I personally wish the codec would just disappear, it's a PITA. 

PCM 5.1 isn't supported over optical or ARC, so it has limited compatibility too, although there are a lot more AVRs that support it than AAC 5.1.  Does the AppleTV support PCM 5.1 passthrough? I thought it was pretty limited when it comes to any lossless surround passthrough?

Kodi is pretty flexible/configurable, I think you can also have it software decode TrueHD to PCM 5.1 if the device doesn't support TrueHD passthrough.  

I would say that AC3 is the most compatible surround codec, and I don't think there is any licencing fee to transcoding to it.  eAC3 would be a better codec, but you lose compatibility, probably requires more processing to transcode, and there may still be a licensing cost?

**Comment by Thlerry** [comment]
Yes the Apple TV supports only Dolby Digital or PCM Multi-channel, and passes Atmos lossy in Dolby Mat 2.0 container.

When "Best Quality is selected" anything which is not Dolby gets decoded on Apple TV and passed as PCM MultiChannel (Unless the option to transcode to Dolby is selected in the sound settings). 

So this would happen for most formats, including lossless Dolby TrueHD and DTS-MA.
The drawback is that for object based format like Dolby Atmos TrueHD and DTS-X MA, you will only get PCM Multi channel at best out of the Apple TV. So it won't be compressed but you lose all the meta data.

**Comment by finnzz** [comment]
I didn't realize the AppleTV was that ridiculous when it came to audio output.  They basically decided to back the most obscure format possible for Atmos.  Dolby Mat 2 does support  PCM with the Atmos meta layer, so I would hope that it could decode TrueHD to multichannel PCM and put it into the Dolby Mat container with the Atmos layer.  That's probably too much to hope for.  

And so all the streaming services that provide Dolby digital plus Atmos have to be decoded to PCM and placed into a mat container with the Atmos layer? Making the output from AppleTV compatible with far fewer soundbars and receivers?

Sometimes I really hate how Apple has to go against the flow just to create more complications with little to no benefit.  It's clear at this point that eAC3-JOC is the standard for Atmos on streaming services.  I think even AppleTVs originals use eAC3-JOC :/
